paf-mvp-frontend/src/lib/paf-lib.ts

```diff
@@ -432,22 +435,117 @@ export const getIdsAndPreferences = (): IdsAndPreferences | undefined => {
   return values as IdsAndPreferences;
 };
 
-export const createSeed = async (
+/**
+ * Aggregate Seed and Preferences for storage.
+ *
+ * We must record the Ids and Preferences
+ * in case they change in the cookies after the
+ * generation of the Seed.
+ */
+interface SeedEntry {
+  seed: Seed;
+  idsAndPreferences: IdsAndPreferences;
+}
+
+type PrebidTransactionId = string;
+type SlotId = string;
+
+/**
+ * Internal storage for generated Seeds.
+ */
+const seedStorage = new Map<TransactionId, SeedEntry>();
+const auditLogByPrebidTransactionId = new Map<PrebidTransactionId, AuditLog>();
+const prebidTransactionIdBySlotId = new Map<SlotId, PrebidTransactionId>();
+
+const createSeed = async (
   { proxyHostName }: CreateSeedOptions,
   transactionIds: TransactionId[]
-): Promise<Seed | undefined> => {
+): Promise<SeedEntry | undefined> => {
   const getUrl = getProxyUrl(proxyHostName);
   const url = getUrl(jsonProxyEndpoints.createSeed);
-  const idsAndPrefs = getIdsAndPreferences();
-  if (idsAndPrefs === undefined) {
+  const idsAndPreferences = getIdsAndPreferences();
+  if (idsAndPreferences === undefined) {
     return undefined;
   }
 
   const requestContent: PostSeedRequest = {
     transaction_ids: transactionIds,
-    data: idsAndPrefs,
+    data: idsAndPreferences,
   };
   const response = await postJson(url, requestContent);
+  const seed = (await response.json()) as Seed;
+
+  return {
+    seed,
+    idsAndPreferences,
+  };
+};
+
+/**
+ * Call hostname endpoint for generating a seed
+ * and store the Seed on Browser side so that it can be retrieved later.
+ *
+ * @param hostname domain called for creating the Seed (POST /paf-proxy/v1/seed).
+ * @param pafTransactionIds List of Transaction Ids for PAF (visible in the Audit).
+ * @returns A new signed Seed that can be used to start Transmissions Requests.
+ */
+export const generateSeed = async (hostname: string, pafTransactionIds: TransactionId[]): Promise<Seed | undefined> => {
+  const entry = await createSeed({ proxyHostName: hostname }, pafTransactionIds);
+
+  if (entry !== undefined) {
+    for (const id of pafTransactionIds) {
+      seedStorage.set(id, entry);
+    }
+  }
+
+  return entry.seed;
+};
 
-  return await response.json();
+/**
+ * Register the Transmission Response of an initial Transmission Request for a given Seed.
+ * @param prebidTransactionId Transaction Id generated by Prebidjs.
+ * @param pafTransactionId  Transaction Id generated for the PAF Audit Log and used for signing the Seed.
+ * @param slotId
+ * @param transmissionResponse Transmission Response of an initial Transmission Request containing all the children.
+ * @returns The generated AuditLog or undefined if the given Transaction Id is unknown or malformed.
+ */
+export const registerTransmissionResponse = (
+  prebidTransactionId: PrebidTransactionId,
+  pafTransactionId: TransactionId,
+  slotId: SlotId,
+  transmissionResponse: TransmissionResponse
```

Comment:
The current API respect the signature designed in the PR on the specifications. However, I think that we need also to add the ContentId of the winning bid so that we can find it in the TransmissionResponse and build the AuditLog.

Reply:
My first instinct was that the method to fetch the audit log would take content id as input. And that, naïvely, only at that time would you determine which branch of transmissions to include in the audit log.

That would support a model where register Transmission Response would only be called once for all slots. Requiring a content id in the input wouldn't support that.

Reply:
After an informal discussion with @broggeri , we keep the method to fetch the audit log without the content id to move forward quickly and we will clarify those API with @bwschmidt .

@OlivierChirouze , I am not really convinced about the simplification of your suggestion. However, I think that it can - indeed - help to refine the API later without a misleading call. So I take it. Thank you.

Reply:
TL;DR There is no such thing as named parameters in typescript ;-)

My point about simplification was that 3 function parameters in a raw with the same type (`string`) is error-prone: you can easily mix them up and write `registerTransmissionResponse("transactIdY", "transactIdX", "slotIdZ")` when you meant `registerTransmissionResponse("transactIdX", "transactIdY", "slotIdZ")`.
With an object and 3 attributes, such error is less likely to happen.

Reply:
Ok, if it is your concern, an anonymous type within the method signature does the job without an extra semantic.
```
export const registerTransmissionResponse = (ctx : { 
  prebidTransactionId: PrebidTransactionId,
  pafTransactionId: TransactionId,
  slotId: SlotId },
  transmissionResponse: TransmissionResponse) {

};
```

Reply:
Yes, but this prevents users of the function to create their objects outside of the function, which I find convenient.
Example:
```javascript
const context: TransmissionContext = {
prebidTransactionId: "12345",
  pafTransactionId: "aqsdf8ze",
  slotId:"14"
} 
registerTransmissionResponse(context);
```
This is why I prefer to avoid anonymous types for "public" methods.